src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/documents/TextBoxView.cs

```diff
@@ -2281,7 +2281,7 @@ private Size FullMeasureTick(double constraintWidth, LineProperties lineProperti
                     endOfParagraph = line.EndOfParagraph;
                 }
             }
-            while (!endOfParagraph && DateTime.Now < stopTime);
+            while (!endOfParagraph && measureStopwatch.Elapsed < stopTimeout);
```

Comment:
do you know how sensitive this code-path is to the accuracy of this measurement? the differences pointed out by SO make me wonder if we could possibly alter behavior that was dependent on the inaccuracy of this: https://stackoverflow.com/questions/2923283/stopwatch-vs-using-system-datetime-now-for-timing-events

Reply:
I don't know how sensitive the code is to the accuracy. I proposed this change not only for increased accuracy, but primarily for increased performance.
Accessing DateTime.Now is much slower than using a stopwatch. And every bit of performance gain we can achieve, especially if it's easy, should be taken, i guess.
I could write a benchmark to show the real difference if you want me to.

Reply:
It'll probably never matter, but `DateTime.Now` has correctness issues when being used for measuring timeouts or implementing timers, as the clock can be changed both manually and automatically (time zone changes esp. scheduled ones like summer/winter time changes). `DateTime.UtcNow` avoids the timezone problem and is faster than `DateTime.Now` but is still suspect to manual clock changes.